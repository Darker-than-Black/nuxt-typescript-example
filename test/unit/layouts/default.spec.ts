import 'jest-localstorage-mock'
import { RouterLinkStub } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Notifications from 'vue-notification'
import vClickOutside from 'v-click-outside'
import VueTestInstance from '~/test/factory/vue-instance'
import DefaultLayout from '~/layouts/default.vue'

describe('layout default.vue', () => {
  const Nuxt = { template: '<p>Nuxt Pages</p>' }
  let wrapper: any

  beforeEach(() => {
    wrapper = VueTestInstance(DefaultLayout, {
      modules: [BootstrapVue, vClickOutside, Notifications],
      options: {
        stubs: { Nuxt }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check set prop "show-aside"', () => {
    expect(wrapper.find('grid-wrapper-stub').attributes().showaside).toBe('true')
  })

  it('check toggle aside', () => {
    const value: boolean = wrapper.vm.showShortNav
    wrapper.vm.toggleVisibleNav()
    expect(wrapper.vm.showShortNav).toBe(!value)
  })

  it('check update LocalStorage', () => {
    expect(localStorage.setItem).not.toHaveBeenLastCalledWith(wrapper.vm.localStorageVar, wrapper.vm.showShortNav)
    expect(localStorage.__STORE__[wrapper.vm.localStorageVar]).toBe(`${wrapper.vm.showShortNav}`)
  })
})
