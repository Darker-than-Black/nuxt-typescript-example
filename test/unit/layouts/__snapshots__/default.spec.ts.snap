// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout default.vue should match the snapshot 1`] = `
<div>
  <b-header-stub></b-header-stub>
  <grid-wrapper-stub showaside="true" class="mt-3"></grid-wrapper-stub>
  <div>
    <notifications-stub group="" width="300" position="bottom right" classes="notification" animationtype="css" animation="[object Object]" animationname="vn-fade" speed="500" cooldown="0" duration="5000" delay="0" max="Infinity" closeonclick="true"></notifications-stub>
  </div>
</div>
`;
